ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -213,22 +213,31 @@ public void expireData(TableRuntime tableRuntime) {
       if (!expirationConfig.isValid(field, table.name())) {
         return;
       }
-      Instant startInstant;
-      if (expirationConfig.getSince() == DataExpirationConfig.Since.CURRENT_TIMESTAMP) {
-        startInstant = Instant.now().atZone(getDefaultZoneId(field)).toInstant();
-      } else {
-        startInstant =
-            Instant.ofEpochMilli(fetchLatestNonOptimizedSnapshotTime(table))
-                .atZone(getDefaultZoneId(field))
-                .toInstant();
-      }
 
-      expireDataFrom(expirationConfig, startInstant);
+      expireDataFrom(expirationConfig, expireBaseOnRule(expirationConfig, field));
     } catch (Throwable t) {
       LOG.error("Unexpected purge error for table {} ", tableRuntime.getTableIdentifier(), t);
     }
   }
 
+  protected Instant expireBaseOnRule(
+      DataExpirationConfig expirationConfig, Types.NestedField field) {
+    if (expirationConfig.getBaseOnRule() == DataExpirationConfig.BaseOnRule.CURRENT_TIME) {
+      return Instant.now().atZone(getDefaultZoneId(field)).toInstant();
+    } else {
+      Snapshot snapshot = IcebergTableUtil.getSnapshot(getTable(), false);
+      long currentSnapshotTs =
+          Optional.ofNullable(snapshot).isPresent()
+              ? snapshot.timestampMillis()
+              : System.currentTimeMillis();
+      // snapshot timestamp should be UTC
+      return Instant.ofEpochMilli(
```

Comment:
If there is only an "optimizing" snapshot, will the date of this snapshot expire?

Reply:
Yes. If there is no non-optimized snapshot, the TableMaintainer will chose the timestamp of  currentSnapshot as the expire time

Reply:
If the rule is LAST_COMMIT_TIME and non-optimized snapshots do not exist, should file expiration not be enforced?

Reply:
The purpose of keeping a snapshot of the last commit is to preserve historical data, so you're right!